reportek/core/permissions/base.py

```diff
@@ -25,3 +42,49 @@ def has_permission(self, request, view):
 
     def has_object_permission(self, request, view, obj):
         return obj.finalized or super().has_permission(request, view)
+
+
+class EffectiveObjectPermissions(DjangoObjectPermissions):
+
+    def get_required_permissions(self, method, model_cls):
+        perms = super().get_required_permissions(method, model_cls)
+        debug(f'{model_cls.__name__} req. perms: {perms}')
+        return perms
+
+    def get_required_object_permissions(self, method, model_cls):
+        perms = super().get_required_object_permissions(method, model_cls)
+        debug(f'{model_cls.__name__} req. obj. perms: {perms}')
+        return perms
+
+    def has_object_permission(self, request, view, obj):
+        # authentication checks have already executed via has_permission
+        queryset = self._queryset(view)
+        model_cls = queryset.model
+        # user = request.user
+        eff_groups = request.user.effective_groups
+
+        perms = self.get_required_object_permissions(request.method, model_cls)
+
+        eff_perms = get_effective_obj_perms(eff_groups, obj)
+
+        print(perms, eff_perms)
```

Comment:
dibag?

Reply:
Aye. Removed. Speaking of debug prints, here's a gem: https://github.com/gruns/icecream